r/tools/nixlibs.R

```diff
@@ -574,6 +574,8 @@ build_libarrow <- function(src_dir, dst_dir) {
         env_var_list <- c(env_var_list, setNames("BUNDLED", env_var))
       }
     }
+    # We also _do_ want to enable S3, GCS, and ZSTD by default
+    env_var_list <- c(env_var_list, c(ARROW_S3 = "ON", ARROW_GCS = "ON", ARROW_WITH_ZSTD = "ON"))
```

Comment:
I think we want a way to turn these off by setting the env vars, as you would any other feature.

I also wonder whether GCS is worth including.

```suggestion
    # We also _do_ want to enable S3, GCS, and ZSTD by default
    # so that binaries built on CRAN from source are fully featured
    env_var_list <- c(
      env_var_list,
      ARROW_S3 = Sys.getenv("ARROW_S3", "ON"),
      ARROW_GCS = Sys.getenv("ARROW_GCS", "ON"),
      ARROW_WITH_ZSTD = Sys.getenv("ARROW_WITH_ZSTD", "ON")
    )
```